core/storage/sqlite3_ondisk.rs

```diff
@@ -1170,28 +1271,45 @@ pub fn begin_write_wal_header(io: &Rc<dyn File>, header: &WalHeader) -> Result<(
     Ok(())
 }
 
-/*
-    Checks if payload will overflow a cell based on max local and
-    it will return the min size that will be stored in that case,
-    including overflow pointer
-*/
+/// Checks if payload will overflow a cell based on the maximum allowed size.
+/// It will return the min size that will be stored in that case,
+/// including overflow pointer
 pub fn payload_overflows(
     payload_size: usize,
-    max_local: usize,
-    min_local: usize,
+    payload_overflow_threshold_max: usize,
+    payload_overflow_threshold_min: usize,
     usable_size: usize,
 ) -> (bool, usize) {
-    if payload_size <= max_local {
+    if payload_size <= payload_overflow_threshold_max {
         return (false, 0);
     }
 
-    let mut space_left = min_local + (payload_size - min_local) % (usable_size - 4);
-    if space_left > max_local {
-        space_left = min_local;
+    let mut space_left = payload_overflow_threshold_min
```

Comment:
I don't understand this calculation and it should be commented

Reply:
Added a reference to sqlite source in https://github.com/tursodatabase/limbo/pull/539/commits/c417fe788083f2dfa20970c9954facaabd9cf654